### PR TITLE
RHINENG-8133: remove v1 and v2 compatibility from /systems

### PR DESCRIPTION
**Pull request by Dugowitch**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: manager/controllers/utils.go [hunk 1]
```diff
@@ -507,28 +507,14 @@ func systemDBLookups2SystemItems(systems []SystemDBLookup) ([]SystemItem, int, m
 			utils.LogDebug("err", err.Error(), "inventory_id", systems[i].ID, "system groups parsing failed")
 		}
 		data[i] = SystemItem{
-			Attributes: systems[i].SystemItemAttributesAll,
```
Comment (psegedy) [inline manager/controllers/utils.go:527]: shouldn't this stayed as `SystemItemAttributesAll`? We probably need to return more attributes similarly to SystemDetail handler

**Review by psegedy — CHANGES_REQUESTED** [review]

File: manager/controllers/systems_ids_test.go [hunk 2]
```diff
@@ -137,14 +137,6 @@ func TestSystemsIDsFilterNotExisting(t *testing.T) {
...
 }
 
-func TestSystemsIDsFilterPartialOS(t *testing.T) {
-	outputIDs := testSystemsIDs(t, "?filter[osname]=RHEL&filter[osmajor]=8&filter[osminor]=1", 1)
-	output := testSystems(t, "?filter[osname]=RHEL&filter[osmajor]=8&filter[osminor]=1", 1)
-	assert.Equal(t, 2, len(outputIDs.IDs))
-	assert.Equal(t, output.Data[0].ID, outputIDs.IDs[0])
-	assert.Equal(t, output.Data[1].ID, outputIDs.IDs[1])
-}
-
```
[the remark this answers is not in the document]
Comment (Dugowitch) [reply manager/controllers/systems_ids_test.go]: Because it tests fields I thought were v2 only

**Review by psegedy — APPROVED** [review]
LGTM! Great job

File: manager/controllers/utils.go [hunk 3]
```diff
@@ -74,12 +73,7 @@ func ApplySort(c *gin.Context, tx *gorm.DB, fieldExprs database.AttrMap,
 
 	// We sort by a column expression and not the column name. The column expression is retrieved from fieldExprs
 	for _, enteredField := range fields {
-		origEnteredField := enteredField // needed for showing correct info in `meta` section
-		if apiver < 3 && strings.Contains(enteredField, "applicable_systems") {
-			// `applicable_systems` is used in v1 and v2 API for consistency, in fact it is installable_systems
-			// therefore, if user wants to sort by `applicable_systems`, we need to sort `installable_systems` in DB
-			enteredField = strings.Replace(enteredField, "applicable", "installable", 1)
-		}
+		origEnteredField := enteredField                                               // needed for showing correct info in `meta` section
```
Comment (psegedy) [inline manager/controllers/utils.go:76]: some more tabs were added and they are the reason why you had to add nolint:lll on line 60, please remove those tabs and nolint ignore for lll

Comment (psegedy) [reply manager/controllers/utils.go:76]: it's done by gofmt

Comment (Dugowitch) [reply manager/controllers/utils.go:76]: it's caused by gofmt

**Pull request merged**